salt/runners/salt.py

```diff
@@ -85,7 +86,13 @@ def cmd(fun, *args, **kwargs):
     with_pillar = kwargs.pop('with_pillar', False)
 
     opts = copy.deepcopy(__opts__)
-    opts['grains'] = salt.loader.grains(opts)
+    # try to only load grains if we need to, it may already exist from other contexts (ie pillar)
```

Comment:
Are pillars the only place where the grains could come from? If there are other places, [e.g.](https://www.grammarly.com/blog/know-your-latin-i-e-vs-e-g/) would be more appropriate.

Reply:
any jinja render, though pillar is the only one i can think of that provides a stubbed grain context. i'll update the comment though